ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/interceptor/Interceptor.java

```diff
@@ -0,0 +1,189 @@
+ *
+ * @since 1.9.0
+ */
+public abstract class Interceptor extends Service {
+
+    public enum InterceptorType {
+        REQUEST, REQUEST_ERROR, RESPONSE, RESPONSE_ERROR
+    }
+
+    private final String name;
+    protected final ClassSymbol serviceClass;
+    private ClassDefinitionNode serviceClassNode = null;
+    protected ResourceMethodSymbol resourceMethod = null;
+    protected boolean continueExecution = false;
+    protected boolean hasNilReturn = false;
+    private Interceptor nextInReqPath = null;
+    protected Interceptor nextInResPath = null;
+
+    protected Interceptor(TypeReferenceTypeSymbol typeSymbol, SemanticModel semanticModel,
+                       ModuleMemberVisitor moduleMemberVisitor) {
+        super(semanticModel);
+        this.name = typeSymbol.getName().orElse("");
+        if (typeSymbol.getName().isPresent() &&
+                moduleMemberVisitor.getInterceptorServiceClassNode(this.name).isPresent()) {
+            this.serviceClassNode = moduleMemberVisitor.getInterceptorServiceClassNode(this.name).get();
+        }
+        this.serviceClass = typeSymbol.typeDescriptor() instanceof ClassSymbol ?
+                (ClassSymbol) typeSymbol.typeDescriptor() : null;
+        extractInterceptorDetails(semanticModel);
+    }
+
+    protected abstract void extractInterceptorDetails(SemanticModel semanticModel);
+
+    public abstract boolean isNotInvokable(ResourceMethodSymbol targetResource);
+
+    protected void setReturnType(TypeSymbol returnType) {
+        hasNilReturn = semanticModel.types().NIL.subtypeOf(returnType);
+        TypeSymbol effectiveReturnType;
+        if (isSubTypeOfDefaultInterceptorReturnType(returnType, semanticModel)) {
+            continueExecution = true;
+            effectiveReturnType = getEffectiveReturnType(returnType, semanticModel);
+        } else {
+            effectiveReturnType = returnType;
+        }
+
+        if (Objects.isNull(effectiveReturnType)) {
+            return;
+        }
+
+        extractErrorAndNonErrorReturnTypes(effectiveReturnType);
+    }
+
+    private boolean isSubTypeOfDefaultInterceptorReturnType(TypeSymbol typeSymbol, SemanticModel semanticModel) {
+        Optional<Symbol> optNextServiceType = semanticModel.types().getTypeByName("ballerina", "http",
+                "", "NextService");
+        if (optNextServiceType.isEmpty() ||
+                !(optNextServiceType.get() instanceof TypeDefinitionSymbol nextServiceType)) {
+            return false;
+        }
+        UnionTypeSymbol defaultInterceptorReturnType = semanticModel.types().builder().UNION_TYPE.withMemberTypes(
+                nextServiceType.typeDescriptor(), semanticModel.types().NIL).build();
+        return defaultInterceptorReturnType.subtypeOf(typeSymbol);
+    }
+
+    private boolean isSubTypeOfHttpNextServiceType(TypeSymbol typeSymbol, SemanticModel semanticModel) {
+        Optional<Symbol> optNextServiceType = semanticModel.types().getTypeByName("ballerina", "http",
```

Comment:
```suggestion
        Optional<Symbol> optNextServiceType = semanticModel.types().getTypeByName("ballerina", "http",
```
seems u have missed these

Reply:
Will update. Thanks 👍🏼

Reply:
Fixed via [f943717](https://github.com/ballerina-platform/openapi-tools/pull/1614/commits/f9437176aa82cf6a737db7ab0301feca63ed985f)